helix-core/src/main/java/org/apache/helix/HelixManager.java

```diff
@@ -425,6 +426,20 @@ void addExternalViewChangeListener(org.apache.helix.ExternalViewChangeListener l
    */
   boolean isLeader();
 
+  /**
+   * Checks whether the cluster manager is leader and sets its ZK session in param
+   * {@link InstanceLeaderSession} if and only if it is leader.
+   *
+   * @param instanceLeaderSession To include ZK session ID of the cluster manager in return
+   *
+   * @return true if the instance is a leader of the cluster and Zk session of the cluster
+   * manager is return in param {@link InstanceLeaderSession}
+   */
+  default boolean isInstanceLeader(InstanceLeaderSession instanceLeaderSession) {
+    throw new UnsupportedOperationException(
+        "Checking leader and returning session is not supported.");
+  }
```

Comment:
I was making it the same name `isLeader(session)`. @lei-xia reminded that helix manager does not have a controller concept so no leader/follower concept. I thought the instance itself managed by the helix manager is a controller. So I name it `isInstanceLeader(session)`. We could discuss and have an agreement. @jiajunwang Could you also take a look? Thanks.

Reply:
@dasahcc An update. I've synced up with @jiajunwang about this. We decided to change it a bit to `Optional<String> getSessionIdIfLeader()` to avoid the new wrapper `InstanceLeaderSession`.